src/controllers/dashboard/plugins/add/index.js

```diff
@@ -65,6 +65,11 @@ function renderPackage(pkg, installedPlugins, page) {
 
     $('#description', page).text(pkg.description);
     $('#developer', page).text(pkg.owner);
+    // This is a hack; the repository name and URL should be part of the global values
+    // for the plugin, not each individual version. So we just use the top (latest)
+    // version to get this information.
+    $('#repositoryName', page).text(pkg.versions[0].repositoryName);
+    $('#repositoryUrl', page).text(pkg.versions[0].repositoryUrl);
```

Comment:
This can fail if the manifest does not have any versions

Reply:
I also can't see a (valid) scenario where there's a plugin entry in a repo with zero versions, and then at worst this would throw and error and be empty, so I think it's fine.

Reply:
I don't it's fine because the webui might become unresponsive. I believe the server only returns versions with a compatible targetAbi so it is possible to not receive any versions.

You can check if there is at least one version before setting the properties like so;

```suggestion
if (pkg.versions.length) {
    $('#repositoryName', page).text(pkg.versions[0].repositoryName);
    $('#repositoryUrl', page).text(pkg.versions[0].repositoryUrl);
 }
```